### ISSUE TITLE
Bar chart doesn't handle multiple series

### ISSUE THREAD

**Issue by dceejay** [node-red/node-red-dashboard#224]
The Chart widget in Bar graph mode handles each "topic" as a different series - so they all get a a different colour... not a bad default. But would be better if they could handle multiple series - where each series has it's own colour. This would let people do things either way - either all have the data one colour (same series, different topic) , or different colours (same topic, different series), or indeed a combination.

This issue is to discuss, and track progress

**Comment by dceejay** [comment]
The underlying chart.js formats do allow this, but the move from a previous library meant we tried to keep the input/output format the same which then won't let us "get at" that format easily as it had no concept of this. 

The proposal is to add a checkbox to select "new behaviour", which will then break the existing save data object format, and replace it with the raw chart.js data array - which will allow much richer arrays to be pushed. 

At the same time - update the widget to accept msg.series  (as well as msg.topic - where topic -> label in chart.js language). So we can do this
![image](https://cloud.githubusercontent.com/assets/5375409/25796630/2e9df748-33d2-11e7-9a78-f0e9544c235a.png)

Very much work in progress

**Comment by nygma2004** [comment]
Interesting. Thanks for letting me know in the group. I will update my code if the new version comes out.

A slightly unrelated question: does chart.js support output? E.g. can the node have another output which provides the image of the chart? I was trying to implement a function where I send the chart in email as an attached png. I could do it with plotly, but that required the chart to be rendered on the screen (for the script to trigger), which meant a second - slightly different looking - chart appeared on the screen. And it only worked if the user was in the UI. So overall mixed results.

**Comment by dceejay** [comment]
the chart outputs the current data array - which I am proposing changes to the format above. 

Interesting thought re image output - but to be useful this would need to render on the server side rather than the browser side, where it is today. Either that or the window would need to be left open and running and sent back as you point out... so not in any current plans.

**Comment by shrickus** [comment]
Last week I was looking at another potential option -- let me know what you think of this.

Currently, there are two "modes" of input, one point (number) at a time, or a set (array) of numbers for historical data. In your example data above, you have an array of 1 object, where "values" is an object containing all the graph data (+ labels, series, etc.)

If you added another mode looking for msg.payload to be an "object", could you just replace the "values" field with the new data and re-render the graph? Granted, the msg.payload would need to be constructed to match whatever input ChartJS needed, but this would allow the flow writer to use all the options available in the charting library. And as the library grows, the node editor would not need to expose all the additional options added to existing chart types.

Here is the data that I was trying to inject into the ui_chart node:
```
  data: {
    labels: ["-7", "-6", "-5", "-4", "-3", "-2", "-1"],
    datasets: [{
      label: 'kWh',
      data: [12, 19, 3, 17, 28, 24, 7],
      backgroundColor: "rgba(153,255,51,.8)"
    }]
  }
```
This was from [this codepen](http://codepen.io/anon/pen/xddeaZ) by Andreas Fuchs on May 1, which you can see renders just fine in the chart library.

**Comment by dceejay** [comment]
Sort of... though we use the angular chart.js library, rather than the pure chart.js - so it would have to be that format instead. http://jtblin.github.io/angular-chart.js/
one step at a time - but def something to think about... maybe rather than that being the object the store/replace payload should be the object... then an array for multiple points to add.. and a simple value to add a single point.  hmm my brain hurts.

**Comment by jroux1** [comment]
Looking forward to this, which ever way you guys are going !

**Comment by dceejay** [comment]
closed by 5d20ac6f4fe37ea25201a41172f8cd01490e16a2
